kie-api/src/test/java/org/kie/api/internal/utils/ServiceDiscoveryImplTest.java

```diff
@@ -16,30 +16,77 @@
 
 package org.kie.api.internal.utils;
 
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
 import java.util.Map;
 
 import org.junit.Test;
 import org.kie.api.internal.assembler.KieAssemblerService;
 import org.kie.api.io.ResourceType;
 
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
 
 public class ServiceDiscoveryImplTest {
 
     @Test
     public void testServiceAndChildServiceInSameKieConf() {
-        final ServiceDiscoveryImpl serviceDiscovery = ServiceDiscoveryImpl.getInstance();
-        final Map<String, Object> services = serviceDiscovery.getServices();
+        ServiceDiscoveryImpl serviceDiscovery = new ServiceDiscoveryImpl();
+        ClassLoader cl = ServiceDiscoveryImplTest.class.getClassLoader();
+        serviceDiscovery.registerConfs( cl, getUrl( cl, "META-INF/kie.conf.test0" ) );
+
+        Map<String, List<Object>> services = serviceDiscovery.getServices();
         assertTrue(services.size() == 1);
 
-        final Object service = services.get("org.kie.api.internal.assembler.KieAssemblers");
+        Object service = services.get("org.kie.api.internal.assembler.KieAssemblers").get(0);
         assertNotNull(service);
         assertTrue(service instanceof MockAssemblersImpl);
 
-        final Map<ResourceType, KieAssemblerService> childServices = ((MockAssemblersImpl) service).getAssemblers();
+        Map<ResourceType, KieAssemblerService> childServices = ((MockAssemblersImpl) service).getAssemblers();
         assertTrue(childServices.size() == 1);
         assertNotNull(childServices.get(ResourceType.DRL));
         assertTrue(childServices.get(ResourceType.DRL) instanceof MockChildAssemblerService);
     }
+
+    @Test
+    public void testDuplicatedServiceShouldFail() {
+        ServiceDiscoveryImpl serviceDiscovery = new ServiceDiscoveryImpl();
+        ClassLoader cl = ServiceDiscoveryImplTest.class.getClassLoader();
+
+        try {
+            serviceDiscovery.registerConfs( cl, getUrl( cl, "META-INF/kie.conf.test1" ) );
+            serviceDiscovery.registerConfs( cl, getUrl( cl, "META-INF/kie.conf.test2" ) );
+            serviceDiscovery.getServices();
+            fail( "Trying to load a duplicated service should fail" );
+        } catch (Exception e) {
+            System.out.println( e.getMessage() );
+        }
```

Comment:
Why this `try catch` instead of just `throws Exception`?

Reply:
If I throw the Exception the test will fail, while in that case the throwing of an Exception is the expected behaviour and indeed the test fail if the Exception is NOT thrown.

Reply:
@mariofusco 
being that the case, would it be possible to use the form

`@Test(expected = Exception.class)` (better to use a more specific exception)

?
This is what I'm using in PMML to make clear that I'm expecting a test to throw an exception - eventually even a specific Exception (my code almost always throw specific exception, so a generic/unexpected one should be marked as a test failure)